test/visit.cpp

```diff
@@ -32,21 +46,97 @@ class visit_test
     };
 
     void
-    testVisit()
+    testVisitMutable()
+    {
+        {
+            json::value v = value(nullptr);
+            BOOST_TEST(visit(check_mutable{kind::null}, v));
```

Comment:
Why did we bloat the number of lines?

Reply:
Because we have to prove the correct path is taken when passing lvalues.
Previously the test was passing rvalues.

Reply:
Yepp... I see what you mean now :)